use current timestamp instead of finished monotonic time

hophacker opened the pull request:
Normally, a developer expects a data point being inserted into influxdb with a timestamp of the current time. However, this project inserts a timestamp of the finished monotonic time which gets passed from rails' `ActiveSupport:: Notifications `([code](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/notifications.rb)). 

This confused me a lot when I first received lots of timestampes like `717690.570250131`.... I calculated with lots of ways to figure out what does that timestamp mean, and finally managed to know that those timestamps are very close to the time since I reboot my mac. Then I searched and realized that `time since last reboot` has a term called `monotonic time`. I digged into rails source code and got to know that was happening.

I don't know what was the exact purpose of putting finished `monotonic time` into influxdb for timestamps, or it was just a small mistake.

Code fixed upon my issue, and tests all passed.

ChrisBr commented:
Hmm, that's interesting 🤔 I will dig into it ... Thanks for the PR!

hophacker commented:
Sure, I'm glad to make further changes

ChrisBr commented:
@hophacker I'm not sure were you think that the monotonic time is used 🤔 We use everywhere ``subscribe`` which uses ``monotonic: false`` (https://github.com/rails/rails/blob/master/activesupport/lib/active_support/notifications.rb#L235). Also it seems the term monotonic time in Rails context means not the time since rebooted but using an internal, more precise clock:

> A monotonic clock is a time source that won't ever jump forward or backward (due to NTP or Daylight Savings Time updates).

So not sure what your issue here is but I think our implementation is correct 🤔We should not use `current_timestamp` as this can be a lot later than the actual `finished` timestamp.

I also just tried this out in the specs and the correct current time gets written.

Could you maybe add some debug code here: 
https://github.com/influxdata/influxdb-rails/blob/master/lib/influxdb/rails/middleware/simple_subscriber.rb#L40
(like a 'puts finished`) and run `bundle exec rspec spec/integration/metrics_spec.rb`.

Long story, short: I do think the implementation is correct but happy to help debugging what is going on on your machine.

ChrisBr commented on spec/unit/middleware/request_subscriber_spec.rb:
```diff
@@ -39,10 +39,11 @@
       it "sends metrics with taggings and timestamps" do
         expect_any_instance_of(InfluxDB::Client).to receive(:write_point).with(
-          "rails.controller", data.deep_merge(values: { value: 2000 })
+          "rails.controller", data.deep_merge(values: { value: 2000 }, timestamp: InfluxDB::Rails.current_timestamp)
```
This is a source for flaky specs. Although this may succeed on your machine, it will be very flaky in CI as the time you merge in here won't be the same which got reported to `write_point` as it happened already a few `ms` before. You would need to freeze the time here.

hophacker replied:
All of the test failures are due to two cases
* syntax check failures of ruby code, I've corrected those failures and squashed into last commit
* default version of sqlite is `1.4.1`, which causes rails `5.0.x` and `4.2.x` to fail. I've fixed the version to be around `1.3.6` at the second commit. This will also let other PRs down if making requests of travis for them again.

hophacker commented:
Sure, I'll check it out

ChrisBr commented:
Any update @hophacker ?

hophacker commented:
No any update, my last response was:
> All of the test failures are due to two cases

> syntax check failures of ruby code, I've corrected those failures and squashed into last commit
> default version of sqlite is 1.4.1, which causes rails 5.0.x and 4.2.x to fail. I've fixed the version to be around 1.3.6 at the second commit. This issue will also let other PRs down if making requests of travis for them again.

And I'm waiting for this PR to be merged or to be commented for further changes.

ChrisBr commented:
> And I'm waiting for this PR to be merged or to be commented for further changes.

See my comment here: https://github.com/influxdata/influxdb-rails/pull/68#issuecomment-508266187

Long story short: I doubt that our current code is wrong, please double check.

ChrisBr commented:
No response, closing
